src/BLSPublicKeyCompendium.sol

```diff
@@ -33,6 +38,9 @@ contract BLSPublicKeyCompendium is IBLSPublicKeyCompendium {
         BN254.G2Point memory pubkeyG2
     ) external {
         bytes32 pubkeyHash = BN254.hashG1Point(pubkeyG1);
+        require(
+            pubkeyHash != ZERO_PK_HASH, "BLSPublicKeyCompendium.registerBLSPublicKey: cannot register zero pubkey"
+        );
```

Comment:
seems odd we weren't doing this before. @gpsanant is there something else preventing someone from registering the zero key? Lack of knowledge of the corresponding private key?
guess we could check that the G1 point doesn't equal the zero point, but this is equivalent 🤷

Reply:
I'm also curious about this. private key for zero point is just 0.
Seems like one could register by using pkG2=sigG1=0 ? But don't understand pairings enough to know whether `e(sigG1,G2)=e(h(m),pkG2)` since I think (?) `e(0,anything) = e(anything,0) = 0` (it's a multilinear map after all)

Reply:
this isn't really a substantive check, it's not needed, it doesn't prevent any attacks

Reply:
so it might not prevent an attack per se, but it would prevent weird behavior, right?
the fact that the apk doesn't change when zero is added to or subtracted from it feels like it could lead to some strange effects. For example, any signature would appear as if you both did and didn't sign it, depending on the interpretation in the context of the code.
I get that it's like using a key that everyone else knows, but it also has this special aspect. Seems safer to avoid the possibility of it happening.

Reply:
> this isn't really a substantive check, it's not needed, it doesn't prevent any attacks

i think it's substantive now, since nowhere else checks this. i got rid of the checks against the zero hash in the pk registry in favor of checking it here